.github/workflows/compile.js.yml

```diff
@@ -0,0 +1,148 @@
+# This workflow will do a clean installation of node dependencies, cache/restore them, build the source code and run tests across different versions of node
+# For more information see: https://help.github.com/actions/language-and-framework-guides/using-nodejs-with-github-actions
+
+name: Build and Compile push
+
+on:
+  push:
+  pull_request:
```

Comment:
This isn't really necessary unless we expect cross-repo pull requests, and even then, we could restrict it to pull requests targeting `master`.